File: Day98/DSA/Coun_Elements_With_Strictly_Smaller_and_greater_Elements .java
Question:- https://leetcode.com/problems/count-elements-with-strictly-smaller-and-greater-elements/description/?envType=problem-list-v2&envId=array

Code:- 

class Solution {
    public int countElements(int[] nums) {

        int mini = Integer.MAX_VALUE;
        int maxi = Integer.MIN_VALUE;


        for(int num : nums) {
            mini = Math.min(mini, num);
            maxi = Math.max(maxi, num);
        }

        int cnt = 0;

        for(int num : nums)
        {
            if(num > mini && num < maxi)
            {
                cnt++;
            } 
        }

        return cnt;

    }
}

File: Day98/DSA/Coun_Elements_With_Strictly_Smaller_and_greater_Elements .cpp
Question:- https://leetcode.com/problems/count-elements-with-strictly-smaller-and-greater-elements/description/?envType=problem-list-v2&envId=array

Code:- 

class Solution {
public:
    int countElements(vector<int>& nums) {

        int mini = INT_MAX;
        int maxi = INT_MIN;

        // find min and max
        for(int num : nums) {
            mini = min(mini, num);
            maxi = max(maxi, num);
        }

        int cnt = 0;

        // count elements strictly between min and max
        for(int num : nums) {
            if(num > mini && num < maxi) {
                cnt++;
            }
        }

        return cnt;
    }
};